mdm/api/src/main/java/com/gentics/mesh/core/db/Tx.java

```diff
@@ -23,22 +27,32 @@ public interface Tx extends BaseTransaction, DaoCollection, CacheCollection, Sec
 	static ThreadLocal<Tx> threadLocalGraph = new ThreadLocal<>();
 
 	/**
-	 * Set the nested active transaction for the current thread.
+	 * Set the nested active transaction for the current Vert.x context, or thread.
 	 * 
 	 * @param tx
 	 *            Transaction
 	 */
 	static void setActive(Tx tx) {
-		Tx.threadLocalGraph.set(tx);
+		Optional.ofNullable(Vertx.currentContext())
+			.ifPresentOrElse(
+					ctx -> ctx.putLocal("tx", Optional.ofNullable(tx)), 
```

Comment:
The actual change. There are `get()` and `getLocal()` in the Vert.x `Context`, which seem to be functionally equal, but `getLocal()` has internal usages.